app/models/letter_opener_web/letter.rb

```diff
@@ -56,17 +56,19 @@ def attachments
     end
 
     def delete
-      FileUtils.rm_rf("#{LetterOpenerWeb.config.letters_location}/#{id}")
+      return unless valid?
+
+      FileUtils.rm_rf(base_dir.to_s)
     end
 
-    def exists?
-      File.exist?(base_dir)
+    def valid?
+      exists? && base_dir_within_letters_location?
     end
 
     private
 
     def base_dir
-      "#{LetterOpenerWeb.config.letters_location}/#{id}"
+      LetterOpenerWeb.config.letters_location.join(id).cleanpath
```

Comment:
thought: `LetterOpenerWeb.config.letters_location` is actually an instance of `Pathname` so we can leverage its powers 🙂